.github/workflows/build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build and Test React
+
+on:
+    push:
+
+    pull_request:
+        branches: ['main']
+
+    workflow_dispatch:
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v3
+
+            - name: Setup Node
+              uses: actions/setup-node@v3
+              with:
+                  cache: 'npm'
+
+            - name: Install Packages
+              run: npm ci
+
+              # Build before format so that if errors exist in both the error that shows up is a build error and not a format error
+            - name: Build app
+              run: npm run build
+
+            - name: ESLint check
```

Comment:
Do you think linting and formatting should be in a separate "test" job? this might improve readability, but would waste time while spinning up a second runner...

Reply:
Could always add that in later if it's problematic